application/src/main/java/org/thingsboard/server/service/entitiy/alarm/DefaultTbAlarmService.java

```diff
@@ -222,45 +220,53 @@ public AlarmInfo unassign(Alarm alarm, long unassignTs, User user) throws Things
+    }
+
+    private void unassignDeletedUserAlarms(User user) {
+        List<AlarmId> alarmIds = alarmService.findAlarmIdsByAssigneeId(user.getId());
+        for (AlarmId alarmId : alarmIds) {
+            AlarmApiCallResult result = alarmSubscriptionService.unassignAlarm(user.getTenantId(), alarmId, System.currentTimeMillis());
+            Alarm alarm = result.getAlarm();
+            if (!result.isSuccessful()) {
+                continue;
+            }
+            if (result.isModified()) {
+                try {
                     AlarmComment alarmComment = AlarmComment.builder()
-                            .alarmId(alarm.getId())
+                            .alarmId(alarmId)
                             .type(AlarmCommentType.SYSTEM)
-                            .comment(JacksonUtil.newObjectNode().put("text", String.format("Alarm was unassigned because user %s - was deleted",
+                            .comment(JacksonUtil.newObjectNode()
+                                    .put("text", String.format("Alarm was unassigned because user with id %s - was deleted",
                                             (user.getFirstName() == null || user.getLastName() == null) ? user.getName() : user.getFirstName() + " " + user.getLastName()))
                                     .put("userId", user.getId().toString())
                                     .put("subtype", "ASSIGN"))
                             .build();
-                    try {
-                        alarmCommentService.saveAlarmComment(alarm, alarmComment, user);
-                    } catch (ThingsboardException e) {
-                        log.error("Failed to save alarm comment", e);
-                    }
-                    notificationEntityService.logEntityAction(alarm.getTenantId(), alarm.getOriginator(), result.getAlarm(),
-                            alarm.getCustomerId(), ActionType.ALARM_UNASSIGNED, user);
+                    alarmCommentService.saveAlarmComment(alarm, alarmComment, null);
+                } catch (ThingsboardException e) {
+                    log.error("Failed to save alarm comment", e);
                 }
+                notificationEntityService.logEntityAction(alarm.getTenantId(), alarm.getOriginator(), result.getAlarm(),
+                        alarm.getCustomerId(), ActionType.ALARM_UNASSIGNED, null);
             }
-
-        } catch (InterruptedException | ExecutionException | TimeoutException e) {
-            throw new RuntimeException(e);
         }
     }
 
-    @Override
-    public Boolean delete(Alarm alarm, User user) {
-        TenantId tenantId = alarm.getTenantId();
-        notificationEntityService.logEntityAction(tenantId, alarm.getOriginator(), alarm, alarm.getCustomerId(),
-                ActionType.DELETED, user);
-        return alarmSubscriptionService.deleteAlarm(tenantId, alarm.getId());
+    @TransactionalEventListener(fallbackExecution = true)
```

Comment:
Please review your implementation against 
https://bartslota.com/transaction-synchronization-and-spring-events/

Reply:
Listeners make sense for Kafka or Redis communications. Changing the SQL have to be done in the same transaction (without a listener)